src/main/java/mate/academy/intro/service/order/OrderServiceImpl.java

```diff
@@ -0,0 +1,113 @@
+package mate.academy.intro.service.order;
+
+import jakarta.transaction.Transactional;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import mate.academy.intro.dto.order.CreateOrderRequestDto;
+import mate.academy.intro.dto.order.OrderDto;
+import mate.academy.intro.dto.order.OrderItemDto;
+import mate.academy.intro.dto.order.UpdateOrderStatusRequestDto;
+import mate.academy.intro.exception.EntityNotFoundException;
+import mate.academy.intro.mapper.OrderMapper;
+import mate.academy.intro.model.Order;
+import mate.academy.intro.model.OrderItem;
+import mate.academy.intro.model.ShoppingCart;
+import mate.academy.intro.model.Status;
+import mate.academy.intro.repository.order.OrderItemRepository;
+import mate.academy.intro.repository.order.OrderRepository;
+import mate.academy.intro.repository.shoppingcart.ShoppingCartRepository;
+import mate.academy.intro.service.shoppingcart.ShoppingCartService;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final ShoppingCartService shoppingCartService;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemRepository orderItemRepository;
+
+    @Transactional
+    @Override
+    public OrderDto createOrder(Long userId, CreateOrderRequestDto requestDto) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findShoppingCartByUserId(userId);
+        if (shoppingCart.getCartItems().isEmpty()) {

```

Comment:
The method is so huge.
Let's try to refactor it 

```
ShoppingCart cart = cartRepository.findByUserId(userId);
if (cart.getCartItems().isEmpty()) {
        throw new OrderProcessingException("Cart is empty for user: " + userId);
}
Order order = orderMapper.cartToOrder(cart, orderDto.shippingAddress());
cart.clearCart();
return orderMapper.toOrderDto(orderRepository.save(order));
```


Where in OrderMapper class

```
@Mapping(target = "id", ignore = true)
@Mapping(target = "total", source = "cart.cartItems", qualifiedByName = "total")
@Mapping(target = "orderItems", source = "cart.cartItems")
Order cartToOrder(ShoppingCart cart, String shippingAddress);
```

and 

```
@Named("total")
default BigDecimal getTotal(Set<CartItem> cartItems) {
        return cartItems.stream()
                .map(i -> i.getBook().getPrice().multiply(BigDecimal.valueOf(i.getQuantity())))
                .reduce(BigDecimal.ZERO, BigDecimal::add);
}
```

Where in ShoppingCart class

```
public void clearCart() {
        cartItems.clear();
}
```